src/WebJobs.Extensions.OpenAI.AISearch/AISearchProvider.cs

```diff
@@ -0,0 +1,267 @@
+                new SearchField("embeddings", SearchFieldDataType.Collection(SearchFieldDataType.Single))
+                {
+                    VectorSearchDimensions = 1536,
+                    IsSearchable = true,
+                    VectorSearchProfileName = vectorSearchProfile,
+                },
+                new SimpleField("timestamp", SearchFieldDataType.DateTimeOffset) { IsFacetable = true }
+            },
+            SemanticSearch = new()
+            {
+                Configurations =
+                {
+                    new SemanticConfiguration("default", new()
+                    {
+                        ContentFields =
+                        {
+                            new SemanticField("text")
+                        }
+                    })
+                }
+            }
+        };
+
+        await searchIndexClient.CreateIndexAsync(index, cancellationToken);
+    }
+
+    async Task IndexSectionsAsync(SearchClient searchClient, SearchableDocument document, CancellationToken cancellationToken = default)
+    {
+        var iteration = 0;
+        var batch = new IndexDocumentsBatch<SearchDocument>();
+        for (int i = 0; i < document.Embeddings.Response.Value.Data.Count; i++)
+        {
+            batch.Actions.Add(new IndexDocumentsAction<SearchDocument>(
+                IndexActionType.MergeOrUpload,
+                new SearchDocument
+                {
+                    ["id"] = Guid.NewGuid().ToString("N"),
+                    ["text"] = document.Embeddings.Request.Input![i],
+                    ["title"] = Path.GetFileNameWithoutExtension(document.Title),
+                    ["embeddings"] = document.Embeddings.Response.Value.Data[i].Embedding.ToArray() ?? Array.Empty<float>(),
+                    ["timestamp"] = DateTime.UtcNow
+                }));
+            iteration++;
+            if (iteration % 1_000 is 0)
+            {
+                // Every one thousand documents, batch create.
+                IndexDocumentsResult result = await searchClient.IndexDocumentsAsync(batch, cancellationToken: cancellationToken);
+                int succeeded = result.Results.Count(r => r.Succeeded);
+                this.logger.LogDebug("""
+                        Indexed {Count} sections, {Succeeded} succeeded
+                        """,
+                        batch.Actions.Count,
+                        succeeded);
+
+                batch = new();
+            }
+        }
+
+        if (batch is { Actions.Count: > 0 })
+        {
+            // Any remaining documents, batch create.
+            IndexDocumentsResult result = await searchClient.IndexDocumentsAsync(batch, cancellationToken: cancellationToken);
+            int succeeded = result.Results.Count(r => r.Succeeded);
+            string message = $"Indexed {batch.Actions.Count} sections, {succeeded} succeeded";
+            this.logger.LogDebug(message);
```

Comment:
The logging here needs to be updated to match the logging on line 221 - i.e. using log parameters instead of string interpolation.